src/info/info.c

```diff
@@ -24,358 +24,412 @@ Info Handler
-            String *primaryError = strNewFmt("%s: %s", errorTypeName(errorType()), errorMessage());
-            bool primaryMissing = errorType() == &FileMissingError;
-            const ErrorType *primaryErrorType = errorType();
 
-            TRY_BEGIN()
-            {
-                iniLocal = infoLoad(this, storage, fileName, true, cipherType, cipherPass);
-            }
-            CATCH_ANY()
-            {
-                // If both copies of the file have the same error then throw that error,
-                // else if one file is missing but the other is in error and it is not missing, throw that error
-                // else throw an open error
-                THROWP_FMT(
-                    errorType() == primaryErrorType ? errorType() :
-                        (errorType() == &FileMissingError ? primaryErrorType :
-                        (primaryMissing ? errorType() : &FileOpenError)),
-                    "unable to load info file '%s' or '%s" INFO_COPY_EXT "':\n%s\n%s: %s",
-                    strPtr(storagePathNP(storage, fileName)), strPtr(storagePathNP(storage, fileName)),
-                    strPtr(primaryError), errorTypeName(errorType()), errorMessage());
-            }
-            TRY_END();
-        }
-        TRY_END();
-
-        // Load the cipher passphrase if it exists
-        const String *cipherPassSub = iniGetDefault(iniLocal, INFO_SECTION_CIPHER_STR, INFO_KEY_CIPHER_PASS_STR, NULL);
+        INFO_CHECKSUM_SECTION(infoSaveData->checksum, section);
 
-        if (cipherPassSub != NULL)
-            this->cipherPass = jsonToStr(cipherPassSub);
+        ioWrite(infoSaveData->write, BRACKETL_BUF);
+        ioWrite(infoSaveData->write, BUFSTR(section));
+        ioWriteLine(infoSaveData->write, BRACKETR_BUF);
 
-        if (ini != NULL)
-            *ini = iniMove(iniLocal, MEM_CONTEXT_OLD());
+        MEM_CONTEXT_BEGIN(infoSaveData->memContext)
+        {
+            infoSaveData->sectionLast = strDup(section);
+        }
+        MEM_CONTEXT_END();
     }
-    MEM_CONTEXT_END();
+    else
+        INFO_CHECKSUM_KEY_VALUE_NEXT(infoSaveData->checksum);
 
-    FUNCTION_LOG_RETURN(INFO, this);
+    // Save key/value
+    INFO_CHECKSUM_KEY_VALUE(infoSaveData->checksum, key, value);
+
+    ioWrite(infoSaveData->write, BUFSTR(key));
+    ioWrite(infoSaveData->write, EQ_BUF);
+    ioWriteLine(infoSaveData->write, BUFSTR(value));
+
+    FUNCTION_TEST_RETURN_VOID();
 }
 
 /***********************************************************************************************************************************
 Save to file
 ***********************************************************************************************************************************/
 void
-infoSave(
-    Info *this, Ini *ini, const Storage *storage, const String *fileName, CipherType cipherType, const String *cipherPass)
+infoSave(Info *this, IoWrite *write, InfoSaveCallback callbackFunction, void *callbackData)
```

Comment:
Good question, I'm not sure.  All I can figure is in this case the compiler implicitly casts the function type as a pointer (since you obviously can't pass the function itself).

I add * to make it consistent with other places.